### PR TITLE
starting to work on the problem

### PR DESCRIPTION
**Pull request by bfitzpat**
Signed-off-by: bfitzpat@redhat.com <bfitzpat@redhat.com>

### CODE REVIEW

**Comment by bfitzpat** [comment]
@lhein Started this to work on https://issues.redhat.com/browse/FUSETOOLS2-576, so if you get a chance I could use some help diagnosing some of the issues. The JIRA describes the problem and I added a new method that grabs the download path from the github api json that's returned... but that's as far as I got before the EOD Friday

**Comment by bfitzpat** [comment]
The trait test is failing. I suspect something changed in the version, but I honestly don't know enough about the test (or how traits work) to fix it. Any chance I can rope you into looking for a fix @lhein ?

**Comment by lhein** [comment]
I created https://github.com/bfitzpat/vscode-camelk/pull/3 to tackle the broken test and also do some improvements to your other changes. Feel free to review and merge whatever you find useful.

File: src/installer.ts [hunk 1]
```diff
@@ -88,25 +88,38 @@ function updateStatusBarItem(sbItem : vscode.StatusBarItem, text: string, toolti
...
+			extension.setRuntimeVersionSetting(versionToUse);
 		}
 	}
 
 	try {
-		const needsUpdate: boolean = await versionUtils.checkKamelNeedsUpdate(versionToUse);
-		if (needsUpdate) {
-			extension.shareMessageInMainOutputChannel(`Checking to see if Apache Camel K CLI version ${versionToUse} available`);
-		}
+//		if (autoUpgrade) {
```
Comment (lhein) [inline src/installer.ts:117]: commented code?

File: src/installer.ts [hunk 2]
```diff
@@ -115,17 +128,32 @@ export async function installKamel(context: vscode.ExtensionContext): Promise<Er
 	console.log(`Attempting to download Apache Camel K CLI to ${installFolder}`);
 	mkdirp.sync(installFolder);
 
-	const kamelUrl: string = `https://github.com/apache/camel-k/releases/download/${versionToUse}/camel-k-client-${versionToUse}-${platformString}-64bit.tar.gz`;
-	const kamelCliFile: string = `camel-k-client-${versionToUse}-${platformString}-64bit.tar.gz`;
-	const downloadFile: string = path.join(installFolder, binFile);
+	//let kamelUrl `https://github.com/apache/camel-k/releases/download/${versionToUse}/camel-k-client-${versionToUse}-${platformString}-64bit.tar.gz`;
```
Comment (lhein) [inline src/installer.ts:131]: commented code?

File: src/installer.ts [hunk 3]
```diff
@@ -115,17 +128,32 @@ export async function installKamel(context: vscode.ExtensionContext): Promise<Er
...
 
 	const result: boolean = await versionUtils.pingGithubUrl(kamelUrl);
 	if (!result) {
-		var msg = `Camel K CLI Version ${versionToUse} unavailable. Please check the Apache Camel K version specified in VS Code Settings. Inaccessible url: ${kamelUrl}`;
 		extension.shareMessageInMainOutputChannel(msg);
 		throw new Error(msg);
 	}
 
+	const kamelCliFile: string = path.parse(kamelUrl).base;
+	//const kamelCliFile: string = `camel-k-client-${versionToUse}-${platformString}-64bit.tar.gz`;
```
Comment (lhein) [inline src/installer.ts:155]: commented code?

File: src/versionUtils.ts [hunk 4]
```diff
@@ -24,28 +24,33 @@ import * as kamelCli from './kamel';
...
+const LAST_MODIFIED_DATE_OF_DEFAULT_VERSION: string = 'Monday, 27 Jul 2020 19:27:00 GMT';
 let latestVersionFromOnline: string;
 
-async function testVersionAvailable(versionToUse: string): Promise<boolean> {
-	if (versionToUse) {
-		const kamelUrl = `https://github.com/apache/camel-k/releases/download/${versionToUse}/camel-k-client-${versionToUse}-${platformString}-64bit.tar.gz`;
-		return await pingGithubUrl(kamelUrl);
+export async function testVersionAvailable(versionToUse: string): Promise<boolean> {
+	if (platformString && versionToUse) {
+		//const kamelUrl = `https://github.com/apache/camel-k/releases/download/${versionToUse}/camel-k-client-${versionToUse}-${platformString}-64bit.tar.gz`;
```
Comment (lhein) [inline src/versionUtils.ts:38]: commented code?

**Pull request merged**